packages/shared/src/components/modals/post/CreateSharedPostModal.tsx

```diff
@@ -92,8 +109,8 @@ export function CreateSharedPostModal({
   const submitProps = {
     color: ButtonColor.Cabbage,
     variant: ButtonVariant.Primary,
-    disabled: isPosting,
-    loading: isPosting,
+    disabled: isPosting || isCreatingPostModeration,
+    loading: isPosting || isCreatingPostModeration,
```

Comment:
Why are these 2 separate states.
From user perspective they just adding a new post right, don't think you need 2 endpoints for it.

Reply:
@rebelchris For the user the only difference will be the seeing the post vs seeing the pending approval modal.

As for to why there's two endpoints, we had a discussion on it here:
https://github.com/dailydotdev/daily-api/pull/2373#discussion_r1824057034

Reply:
@rebelchris I think you were right and did a rework to more match your suggestion. 

However, I still left the creation of freeform post more or less how it is, since it didn't have its own hook and is handled in the page. I can create one if that is desired, though.